build.assets/tooling/cmd/build-apt-repos/apt_repo_tool.go

```diff
@@ -0,0 +1,178 @@
+func (art *AptRepoTool) Run() error {
+	start := time.Now()
+	logrus.Infoln("Starting APT repo build process...")
+
+	err := art.s3Manager.DownloadExistingRepo(*art.config.localBucketPath)
+	if err != nil {
+		return trace.Wrap(err, "failed to sync existing repo from S3 bucket")
+	}
+
+	repos, err := art.createRepos()
+	if err != nil {
+		return trace.Wrap(err, "failed to create repos")
+	}
+
+	err = art.importNewDebs(repos)
+	if err != nil {
+		return trace.Wrap(err, "failed to import new debs")
+	}
+
+	err = art.publishRepos(repos)
+	if err != nil {
+		return trace.Wrap(err, "failed to publish repos")
+	}
+
+	aptlyRootDir, err := art.aptly.GetRootDir()
+	if err != nil {
+		return trace.Wrap(err, "failed to get Aptly root directory")
+	}
+
+	err = art.s3Manager.UploadBuiltRepo(filepath.Join(aptlyRootDir, "public"))
+	if err != nil {
+		return trace.Wrap(err, "failed to sync changes to S3 bucket")
+	}
+
+	execution_duration := time.Since(start).Round(time.Millisecond)
+	logrus.Infof("APT repo build process completed in %s\n", execution_duration)
+	return nil
+}
+
+func (art *AptRepoTool) publishRepos(repos []*Repo) error {
+	// Build a map keyed by os info with value of all repos that support the os in the key
+	// This will be used to structure the publish command
+	categorizedRepos := make(map[string][]*Repo)
+	for _, r := range repos {
+		if osRepos, ok := categorizedRepos[r.OSInfo()]; ok {
+			categorizedRepos[r.OSInfo()] = append(osRepos, r)
+		} else {
+			categorizedRepos[r.OSInfo()] = []*Repo{r}
+		}
+	}
+
+	for osInfo, osRepoList := range categorizedRepos {
+		if len(osRepoList) < 1 {
+			continue
+		}
+
+		err := art.aptly.PublishRepos(osRepoList, osRepoList[0].os)
+		if err != nil {
+			return trace.Wrap(err, "failed to publish for os \"%s\"", osInfo)
+		}
+	}
+
+	return nil
+}
+
+func (art *AptRepoTool) recreateExistingRepos() ([]*Repo, error) {
+	logrus.Infoln("Recreating previously published repos...")
+	createdRepos, err := art.aptly.CreateReposFromPublishedPath(*art.config.localBucketPath)
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to recreate existing repos")
+	}
+
+	for _, repo := range createdRepos {
+		err := art.aptly.ImportDebsFromExistingRepo(repo)
+		if err != nil {
+			return nil, trace.Wrap(err, "failed to import debs from existing repo \"%s\"", repo.Name())
+		}
+	}
+
+	logrus.Infof("Recreated and imported pre-existing artifacts for %d repos", len(createdRepos))
+	return createdRepos, nil
+}
+
+func (art *AptRepoTool) createRepos() ([]*Repo, error) {
+	logrus.Infoln("Creating Aptly repos...")
+	createdExistingRepos, err := art.recreateExistingRepos()
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to recreate existing repos")
+	}
+
+	createdNewRepos, err := art.aptly.CreateReposFromArtifactRequirements(*art.supportedOSs, *art.config.releaseChannel, *art.config.majorVersion)
+	if err != nil {
+		log.Fatal(err.Error())
```

Comment:
```suggestion
		return nil, trace.Wrap(err)
```

Reply:
Thanks - I missed that when changing out my error handling code